proposals/001-controlled-rollout.md

```diff
@@ -0,0 +1,115 @@
+# Controlled Rollout proposal
+
+## Why
+
+New features are great, everybody loves them. However, new features come by the hand of new code. New code can have bugs 
+or it can have performance regressions.
+
+Features aren't perfect for all users. Sometimes they are perfect for some users and not others. Sometimes they are 
+perfect for some hosts and not others.
+
+Rolling out a feature shouldn't always be a binary choice: enabled/disabled. In an ideal world, all features would be 
+enabled by default, everybody would love them all, and they would work flawlessly for all possible use cases.
+
+We are in the real world, though, which is not ideal. So we should give people running Fleet tools to rollout features 
+slowly, so that they can update infrastructure if needed, or only use a feature within the scope that is useful for 
+them.
+
+This is a proposal on how this tool could look like and work.
+
+## How
+
+We would create a new type of boolean value in our `AppConfig` called `RolloutBoolean`.
+
+`RolloutBoolean` will have a function `Enabled(h *fleet.Host) bool`. So instead of doing this:
```

Comment:
nice! as a general comment, I think it would be useful to include which `RolloutBoolean` values are `true` in the error instance we store in Redis.

Reply:
Could you expand on your comment? I'm not sure I follow.

Reply:
yeah, sorry that was pretty abstract. I meant it would be nice to ultimately include which flags are enabled in `fleetctl debug errors` and/or `fleetctl debug archive`